src/qtgui/iq_tool.cpp

```diff
@@ -303,13 +316,16 @@ void CIqTool::refreshDir()
 {
     int selection = ui->listWidget->currentRow();
 
+    QScrollBar * sc = ui->listWidget->verticalScrollBar();
+    int lastScroll = sc->sliderPosition();
     recdir->refresh();
     QStringList files = recdir->entryList();
 
     ui->listWidget->blockSignals(true);
     ui->listWidget->clear();
     ui->listWidget->insertItems(0, files);
     ui->listWidget->setCurrentRow(selection);
+    sc->setSliderPosition(lastScroll);
```

Comment:
Seems like only the 3 changed lines above are actually fixing the #1037, the others seems improve or fix other possible (but related) issues. Would make sense to split them?

Actually mey-be use of QFileSystemWatcher would be here more appropriate?

Reply:
But actually this can be good start and the `QFileSystemWatcher` can be done possibly latter ...